src/Service/Modules/Reports/PurchaseOrdersReportModule.cs

```diff
@@ -52,5 +67,32 @@ private async Task GetOrdersBySupplierReport(HttpRequest req, HttpResponse res)
 
             await res.Negotiate(results);
         }
+
+        private async Task GetOrdersBySupplierExport(HttpRequest req, HttpResponse res)
+        {
+            var resource = new OrdersBySupplierSearchResource();
+            resource.SupplierId = req.Query.As<int>("Id");
+            resource.From = req.Query.As<string>("FromDate");
+            resource.To = req.Query.As<string>("ToDate");
+
+            resource.Returns = req.Query.As<string>("Returns");
+            resource.Outstanding = req.Query.As<string>("Outstanding");
+            resource.Cancelled = req.Query.As<string>("Cancelled");
+            resource.Credits = req.Query.As<string>("Credits");
+            resource.StockControlled = req.Query.As<string>("StockControlled");
+
+            var results = this.purchaseOrderReportFacadeService.GetOrdersBySupplierExport(resource, req.HttpContext.GetPrivileges());
+
+            this.csvStreamWriter.WriteModel(results);
```

Comment:
It does feel a bit wrong to be doing the IEnumerable<IEnumerable<string>> -> csv stream conversion in the Carter Module but I can't think of a perfect solution off of the top of my head.

Maybe we could have a further Response extension method like FromStreamToCsv in common that makes use of the existing FromStream() one and does the csv convert too

Reply:
I don't mind it too much since it's just calling common, I guess we could rewrite stuff to do it in the facade, and then I guess we'd return a stream from the facade? Not sure that's nice either

Reply:
i'd +1 for the facade or domain service definitely rather than the module.

Reply:
Instead of returning a stream from facade, I could pass it through from the module so I can still use its contents, then within the facade (or domain even) can new up the csvStreamWriter and call WriteModel. Does that sound better to you guys?

Reply:
I guess we could just have a method in the FacadeService that returns a stream as opposed to a list of lists of strings

Reply:
Tried returning the stream, but because we're doing using within the facade now it fails. I guess I'll pass it in and return void or a success result
![image](https://user-images.githubusercontent.com/51612225/151169203-f4253d62-b63c-4e3f-a354-703f8ef326b0.png)

Reply:
Will fix after lunch 😫 
![image](https://user-images.githubusercontent.com/51612225/151170421-1462034a-8a6b-4cd4-9006-398a4b07ce8a.png)

Reply:
still getting that ^, tempted to revert and try work out a facade solution that works in my next pr